### ISSUE TITLE
Better filtering options for Channels with different resolutions.

### ISSUE THREAD

**Issue by jonasthim** [xteve-project/xTeVe#127]
I'd like a clean program list where I don't have the same channels like

-  SomeChannelName HD, 
-  SomeChannelName FHD, 
-  SomeChannelName UHD.

Where I would only have the UHD channel visible for my clients. I know the possibilty exists to filter out like only UHD, but I don't know which channels are available in which format on beforehand.

**Describe the solution you'd like**
Better filter options, where you can specify in the UI a checkbox with "Remove channels with the same name" and a list of filter options with priority, like inputs where you can specify the "UHD"/"FHD"/"HD" - and assign a priority to each row.

**Describe alternatives you've considered**
Go through the EPG list and make 1000 custom filters :) Which seems like a hassle when new channels are added.

Maybe somthing like this,
![image](https://user-images.githubusercontent.com/3851453/80589804-2ce24d80-8a1b-11ea-9c6e-b376b770046c.png)

**Comment by mar-mei** [comment]
The filter is there to trim large playlists down to a few hundred channels. You can deactivate individual channels in the mapping menu.

**Comment by jonasthim** [comment]
Yes, but it would be great with some extra "automation" to this - so you don't have to manually go through the list.

**Comment by alturismo** [comment]
that would mean xteve needs to open the streams and analyse ... which will sure not come ...

either u filter by name and hope your providers naming is correct or ... manually by yourself.

**Comment by tunes0710** [comment]
I'm wondering if people are slightly misunderstanding the request. (Or perhaps I'm misunderstanding the replies).

I believe what @jonasthim is suggesting is...

BBC One SD
BBC One FHD
BBC One UHD

With the priorities set (as in their suggestion above)
SD=1
FHD=5
UHD=10

(Highest Priority is displayed)

In the above situation BBC Ond UHD would be listed as it has the highest priority and SD/FHD would be excluded.

However if it was
BBC One SD
BBC One FHD

Then the FHD channel would be shown.

or if it was just
BBC One SD

Then that one would be shown, as it is the only one matching that channel...

As when you are importing groups, if you filter out FHD and UHD, then there may be some channels that the provider doesn't provide in SD, and only in FHD and/or UHD, so it would be missed out.

This way you are guaranteed to get every channel with the best possible quality.

*** This does make the assumption that the service provider would be naming the channels correctly ***

So essentially, if there is a UHD version, display that... if not, display the FHD, or if neither are available, it will display the SD version.

I like the idea of this suggestion and gets a +1 from me (if it's possible to implement)

**Comment by alturismo** [comment]
may the question, are there alot channels in different resolutions in your playlists ?

i mean disabling 20 channels aint that much effort ...

and this would expect a clean naming schema from the iptv providers, or many many regext and compare rules to get doubles filtered and priorized.

when i look at it, theres not only 1 provider, there are many ... some name BBC ONE HD, the next BBCONE_HD, etc etc ...

and im pretty sure we understand ...

**Comment by tunes0710** [comment]
I did a quick count up in my M3U
SD returned 2896 matches
HD returned 3273 matches
FHD returned 575 matches

But the feature would also be useful so should the provider drop a FHD channel, then it would auto enable the HD or SD channel upon the next m3u update.

It isn't so much about the initial setup, as yes, you could filter out the channels.. but about the ongoing maintenance of checking if a channel has vanished and re-enabling another version of it, whereas with the suggestion above, it would do that on your behalf every time it refreshes the M3U and re-applies the filters.

It was only when it was suggested about analyzing the stream, that I wanted to clarify what I think the OP meant, or at least find out if I had understood it the correct way.

And you are correct, this wouldn't work given your examples above of underscores and different spacing used, but having a quick scan through mine, there are lots of channels where applying the logic above would successfully filter out so the multiple channels are removed (in about 95% of cases, there are still a few channels where the naming convention isn't great, but I've reported these to hopefully be changed on a future update)

**Comment by alturismo** [comment]
in your sample about counting, its not about overall counts, its about doubles

how many channels are in there which are available with different formats like SD, HD, FHD, UHD

in your sceanrio the worst case would be all 232 FHD channels would also be available as HD or SD,
then i would understand that its an effort, but actually i dont believe there are more then 20 or so.

**Comment by tunes0710** [comment]
You also have to remember that this will also include channels that don't have any suffix as well

i.e.

BBC One
BBC One SD
BBC One HD
BBC One FHD

Where no suffix = priority 0
SD = 1
HD = 5
FHD = 10
UHD = 15
etc etc

So when you compare channels after removing suffix to see if they match, and then prioritising by suffix value (if any), then the number of possible channels opens up even more.

I've ran the list through several filters (had to do it in excel, as it's the easiest way for me to do so) and matching using the above logic, I had 235 matches
This doesn't have to be where all of the examples above match it could just be
BBC One (no suffix)
BBC One HD
or
BBC One (no suffix)
BBC One HD
BBC One FHD

As I mentioned earlier... even if we took the time to go through and manually filter the initial list, this feature would still be very handy for ongoing updates to the M3U so we don't have to keep manually checking to see if a HD or FHD channel has disappeared meaning that we then have to go in an enable a no suffix or SD suffixed channel.

**Comment by MCTyler** [comment]
What your asking is too user specific for your one example. Please feel free to fork the Source and edit to your needs for such a specific request.
